Src/Particle/AMReX_ParticleTile.H

```diff
@@ -160,6 +163,60 @@ struct ParticleTileData
     }
 };
 
+// SOA Particle Structure 
+template <int NArrayReal, int NArrayInt>
+struct SoAParticle : SoAParticleBase
+{
+
+    static constexpr bool is_soa_particle = true;
+
+    using RealType = ParticleReal;
+
+    SoAParticle(ParticleTileData<0, 0, NArrayReal, NArrayInt> *const ptd, int const index) 
+    {
+        m_particle_tile_data=ptd;
+        m_index=index;
+    }
+     
+    //functions to get id and cpu in the SOA data
+
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    ParticleCPUWrapper cpu () & { return ParticleCPUWrapper(this->m_particle_tile_data.m_idata[1][m_index]); }
```

Comment:
```suggestion
    ParticleCPUWrapper cpu () & { return ParticleCPUWrapper(this->m_particle_tile_data->m_idata[1][m_index]); }
```
Could you change this (and other places where you use `m_particle_tile_data` from `.` to `->`?

Reply:
@Thierry992